pkg/agent/flowexporter/exporter/exporter_perf_test.go

```diff
@@ -0,0 +1,224 @@
+// +build !race
+
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporter
+
+import (
+	"crypto/rand"
+	"flag"
+	"fmt"
+	"math/big"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/vmware/go-ipfix/pkg/registry"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/flowexporter"
+	"antrea.io/antrea/pkg/agent/flowexporter/connections"
+	"antrea.io/antrea/pkg/agent/flowexporter/flowrecords"
+)
+
+const (
+	testNumOfConns           = 100000
+	testNumOfDenyConns       = 100000
+	testNumOfDyingConns      = 10000
+	testNumOfInactiveRecords = 10000
+	testNumOfIdleDenyConns   = 10000
+	testBufferSize           = 1048
+)
+
+var exp *flowExporter
+
+var count = 0
+
+/*
+Sample output:
+	go test -test.v -run=BenchmarkExport -test.benchmem -bench=. -memprofile memprofile.out -cpuprofile profile.out
+	goos: linux
+	goarch: amd64
+	pkg: antrea.io/antrea/pkg/agent/flowexporter/exporter
+	cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
+	BenchmarkExport
+		exporter_perf_test.go:80:
+			Summary:
+			Number of conntrack connections: 100000
+			Number of dying conntrack connections: 10000
+			Number of deny connections: 100000
+			Number of idle deny connections: 10000
+			Total connections received: 182861
+	BenchmarkExport-2   	       1	5469929206 ns/op	656425016 B/op	17396982 allocs/op
+	PASS
+	ok  	antrea.io/antrea/pkg/agent/flowexporter/exporter	5.914s
+*/
+func BenchmarkExport(b *testing.B) {
+	disableLogToStderr()
+	ctrl := gomock.NewController(b)
+	defer ctrl.Finish()
+	if err := setupExporter(); err != nil {
+		b.Fatalf("error when setting up exporter: %v", err)
+	}
+
+	for n := 0; n < b.N; n++ {
+		exp.Export()
```

Comment:
We are considering one set of connections for one export cycle. This is not exercising any code in AddOrUpdateConn method in the conntrack connection store. Is there a plan to have a test for that too?

Reply:
`AddOrUpdateConn` is part of `Poll()` function which will involve mocking dumping flows output from ovs. Do you mean we can test `AddOrUpdateConn` individually or cover testing for `Poll()` too?

Reply:
Covering the test for `Poll()` sounds good as it would include `AddOrUpdateConn`. However, we may need multiple polls as we want to exercise update code or we should initialize the connection store with some connections before the poll. 

I think we are expecting `Run()` method in exporter.go to be exercised in a different e2e test by running it multiple times. Is that correct?

Reply:
Covering test for `Poll()` --- Sure will add that test.
Benchmarking `Run()` in e2e tests --- Yes we should implement it in the future if you are referring to testing with actual polling, updating and exporting.

Reply:
If possible, let's test things in isolation (when it makes sense) in this type of benchmark